Libraries/LibGUI/Link.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <LibCore/Event.h>
+#include <LibGUI/Event.h>
+#include <LibGUI/Link.h>
+#include <LibGUI/Painter.h>
+#include <LibGfx/Color.h>
+#include <LibGfx/Font.h>
+
+namespace GUI {
+
+void Link::mousedown_event(MouseEvent&)
+{
+    if (on_click) {
+        on_click();
+    }
+}
+
+void Link::paint_event(PaintEvent& event)
+{
+    Label::paint_event(event);
+    GUI::Painter painter(*this);
+    painter.draw_line({ 0, rect().bottom() }, { font().width(text()), rect().bottom() }, Color::MidGray);
```

Comment:
this is wrong as it won't work properly with themes other than the default one, please use `Gfx::Palette` (accessible via `Widget::palette()`)

Reply:
Ah thank you, I wasnt sure what to look for for this!